clusterloader2/pkg/measurement/util/runtimeobjects/runtimeobjects_test.go

```diff
@@ -426,13 +427,111 @@ func changeEnv(in *v1.Pod) *v1.Pod {
 	return out
 }
 
+func noWhitespace(s string) string {
+	s = strings.ReplaceAll(s, " ", "")
+	s = strings.ReplaceAll(s, "\n", "")
+	s = strings.ReplaceAll(s, "\r", "")
+	s = strings.ReplaceAll(s, " ", "") //non-breaking space
+	return strings.ReplaceAll(s, "\t", "")
+}
+
+const fullCompareErrorDifferentEnv = `Not matching templates, diff:   v1.PodSpec{
+    Volumes:        nil,
+    InitContainers: nil,
+    Containers: []v1.Container{
+            {
+                    ... // 5 identical fields
+                    Ports:   nil,
+                    EnvFrom: nil,
+                    Env: []v1.EnvVar{
+                            {
+                                    Name:      "env1",
+-                                   Value:     "val1",
++                                   Value:     "val1-diff",
+                                    ValueFrom: nil,
+                            },
+                    },
+                    Resources:    {Requests: {s"cpu": {i: {...}, Format: "DecimalSI"}, s"memory": {i: {...}, s: "50M", Format: "DecimalSI"}, s"pods": {i: {...}, s: "100", Format: "DecimalSI"}}},
+                    VolumeMounts: nil,
+                    ... // 12 identical fields
+            },
+    },
+    EphemeralContainers:           nil,
+    RestartPolicy:                 "",
+-   TerminationGracePeriodSeconds: nil,
++   TerminationGracePeriodSeconds: &30,
+-   ActiveDeadlineSeconds:         nil,
++   ActiveDeadlineSeconds:         &30,
```

Comment:
Is there a way to ignore fields not set in the first object?

Reply:
Not an easy one. We could vendor this library and do changes in files like https://github.com/google/go-cmp/blob/d103655696d8ae43c4125ee61454dbf03d8e8324/cmp/report.go and https://github.com/google/go-cmp/blob/d103655696d8ae43c4125ee61454dbf03d8e8324/cmp/report_compare.go , however I guess that this would be a little too much work to achieve nicer log. Or we could write our own Reporter, but this is also a significant amount of work (and seeing from documentation not so well supported).